orchagent/vnetorch.cpp

```diff
@@ -1384,17 +1384,21 @@ bool VNetOrch::addOperation(const Request& request)
                 VNetInfo vnet_info = { tunnel, vni, peer_list, scope };
                 obj = createObject<VNetVrfObject>(vnet_name, vnet_info, attrs);
                 create = true;
-            }
 
-            VNetVrfObject *vrf_obj = dynamic_cast<VNetVrfObject*>(obj.get());
-            if (!vxlan_orch->createVxlanTunnelMap(tunnel, TUNNEL_MAP_T_VIRTUAL_ROUTER, vni,
-                                                  vrf_obj->getEncapMapId(), vrf_obj->getDecapMapId(), VXLAN_ENCAP_TTL))
+                VNetVrfObject *vrf_obj = dynamic_cast<VNetVrfObject*>(obj.get());
+                if (!vxlan_orch->createVxlanTunnelMap(tunnel, TUNNEL_MAP_T_VIRTUAL_ROUTER, vni,
+                                                      vrf_obj->getEncapMapId(), vrf_obj->getDecapMapId(), VXLAN_ENCAP_TTL))
+                {
+                    SWSS_LOG_ERROR("VNET '%s', tunnel '%s', map create failed",
```

Comment:
Do we need to clean up the newly created VNET if this tunnel map creation fails? If the VNET gets created but tunnel map creation fails, if the add operation is performed again the tunnel map creation will not be attempted since the VNET already exists.

Reply:
Sure, added a retry handling if map create failed.